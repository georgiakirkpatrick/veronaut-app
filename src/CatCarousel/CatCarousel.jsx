import React from 'react'
import './CatCarousel.css'

const CatCarousel = props => {
  const {
    id,
    images
  } = props

  const formattedImages = images.map(image => {
    return <li key={'li-' + image.id}>
      <div>
        <img
          src={image.url}
          alt={image.category}
        />
      </div>

      <p>{image.category}</p>
    </li>
  })

  return (
    <div 
      id={id}
      className='CatCarousel'
    >
      <ul className='CatCarousel__list'>
        {formattedImages}
      </ul>
    </div>
  )
}

CatCarousel.defaultProps = {
  id: 0,
  images: [
    {
      id: 0,
      url: '',
      category: '',
      slug: ''
    }
  ]
}

export default CatCarousel