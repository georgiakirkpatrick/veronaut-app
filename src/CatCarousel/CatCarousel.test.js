import React from 'react'
import ReactDOM from 'react-dom'
import CatCarousel from './CatCarousel'
import renderer from 'react-test-renderer'

describe ('CatCarousel', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<CatCarousel />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<CatCarousel />)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})