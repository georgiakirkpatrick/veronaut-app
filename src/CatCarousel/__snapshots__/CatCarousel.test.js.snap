// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatCarousel renders the UI as expected 1`] = `
<div
  className="CatCarousel"
  id={0}
>
  <ul
    className="CatCarousel__list"
  >
    <li>
      <div>
        <img
          alt=""
          src=""
        />
      </div>
      <p>
        
      </p>
    </li>
  </ul>
</div>
`;
